client/components/cloud/region/region.service.js

```diff
@@ -18,6 +18,21 @@ class RegionService {
         return number ? number[0] : "";
     }
 
+    getAllTranslatedMacroRegion () {
```

Comment:
Yes this list is hardcoded and should be updated when a new macro region is added.
We must add the corresponding translation before adding a new region, it cannot be generated dynamically.